src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexRangeInAdd(index);
+        checkCapacity();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            checkCapacity();
+            elements[size] = list.get(i);
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final Object removedElement = elements[index];
+        System.arraycopy(elements, index + 1, elements, index, elements.length - index - 1);
+        size--;
+        return (T) removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T removedElement = null;
+        if (element == null) {
+            for (int i = 0; i < this.size; i++) {
+                if (this.elements[i] == null) {
+                    removedElement = remove(i);
+                    return removedElement;
+                }
+            }
+        } else {
+            for (int i = 0; i < this.size; i++) {
+                if (element.equals(this.elements[i])) {
+                    removedElement = remove(i);
+                    return removedElement;
+                }
+            }
+        }
+        if (removedElement == null) {
+            throw new NoSuchElementException("No such element in list");
+        }
+        return removedElement;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkCapacity() {
+        if (size == elements.length) {
+            ensureCapacity();
+        }
+    }
+
+    private void ensureCapacity() {
+        int newCapacity = elements.length * 2;
+        Object[] copyElements = new Object[newCapacity];
+        System.arraycopy(elements, 0, copyElements, 0, elements.length);
+        elements = copyElements;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound list");
```

Comment:
incomplete and invalid error message. Please add more details. Fix the same issue in all places